src/Microsoft.VisualStudio.ProjectSystem.Managed.VS/ProjectSystem/VS/Properties/InterceptedProjectProperties/AbstractBuildEventValueProvider.Helper.cs

```diff
@@ -0,0 +1,136 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+
+using System;
+using System.Linq;
+using Microsoft.Build.Construction;
+
+namespace Microsoft.VisualStudio.ProjectSystem.VS.Properties.InterceptedProjectProperties
+{
+    internal abstract partial class AbstractBuildEventValueProvider
+    {
+        public abstract class Helper
+        {
+            private const string _execTaskName = "Exec";
+            private const string _commandString = "Command";
+
+            protected Helper(string buildEventString,
```

Comment:
`Helper` is a very generic name. This object contains state so I would suggest a better name, like `AbstractBuildEventHelper` or `AbstractBuildEventWorker`